### ISSUE TITLE
RandomAffine3D: scale=(a, b) is documented as isotropic but draws the three axes independently

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4704]
## Summary

`RandomAffine3D` and `AffineGenerator3D` document a two-value `scale` as isotropic:

> scale: scaling factor interval. If (a, b) represents isotropic scaling, the scale is randomly sampled from the range a <= scale <= b. If ((a, b), (c, d), (e, f)), the scale is randomly sampled from the range a <= scale_x <= b, c <= scale_y <= d, e <= scale_z <= f.

`(a, b)` actually draws the three axes **independently**, so a user asking for aspect-preserving scaling gets an anisotropic one. The 2D `RandomAffine` does the documented thing.

```python
import torch, kornia.augmentation as K
for cls, shape in ((K.RandomAffine3D, (2000, 1, 5, 6, 7)), (K.RandomAffine, (2000, 1, 6, 7))):
    s = cls(0.0, scale=(0.5, 2.0), p=1.0).forward_parameters(torch.Size(shape))["scale"]
    print(cls.__name__, int((s.max(1).values == s.min(1).values).sum()), "/ 2000 rows have one scale for every axis")
# RandomAffine3D 0 / 2000      (largest per-row spread 1.48 of the 1.5 available)
# RandomAffine   2000 / 2000

torch.manual_seed(0); aug = K.RandomAffine3D(0.0, scale=(0.5, 2.0), p=1.0); aug(torch.rand(1, 1, 5, 6, 7))
print(aug.transform_matrix[0].diagonal())   # tensor([1.0787, 1.2447, 1.3457, 1.0000])
```

Measured on `main` @ ff204fb9c, torch 2.14.0, CPU float32.

## Cause

`kornia/augmentation/random_generator/_3d/affine.py`: a shape-`(2,)` `scale` is expanded with `_scale.unsqueeze(0).repeat(3, 1)` into three identical **ranges**, and `forward` then draws from `scale_1_sampler`, `scale_2_sampler` and `scale_3_sampler` separately. The 2D generator draws once and repeats the **value** for a two-value range.

## Expected

For `scale=(a, b)`, draw one value per sample and use it on all three axes, as the docstring says and as `RandomAffine` does. `((a, b), (c, d), (e, f))` stays per-axis. This changes the sampled parameters for a given seed, so it needs a changelog entry.

Found while reviewing #4634 (batch 6d of #4407), whose `RandomAffine3D` Convention block now states the current behaviour and cites this issue.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
